CRM/MembersOnlyEvent/Hook/Copy/Event.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+use CRM_MembersOnlyEvent_BAO_EventMembershipType as  EventMembershipType;
+use CRM_MembersOnlyEvent_BAO_MembersOnlyEvent as MemberOnlyEvent;
+
+/**
+ * Class for for Copy Hook for Event object
+ */
+class CRM_MembersOnlyEvent_Hook_Copy_Event {
+
+  /**
+   * Handle Hook Pre Event
+   *
+   * @param  string $objectName
+   * @param  $object
+   * @throws CRM_Core_Exception
+   */
+  public function handle($objectName, &$object) {
+    if (!$this->shouldHandle($objectName)) {
+      return;
+    }
+    $this->createMemberOnlyEventFromEventTemplate($object);
+  }
+
+  protected function createMemberOnlyEventFromEventTemplate(&$objectRef) {
```

Comment:
the only part of this object you are using the Id, so maybe you can just pass it alone to this class. see my other comment for an exmaple

Reply:
@omarabuhussein  Agree to make change for this.